lib/mysqlDaoQueryHelper.js

```diff
@@ -58,6 +58,23 @@ class MysqlDaoQueryHelper {
     }
 
     /**
+     * @param {object} upsertArgs
+     * @returns {string}
+     */
+    upsert(upsertArgs) {
+        const cleanValues = this._cleanAndMapValues(upsertArgs);
+        let insert =
+            Squel.useFlavour('mysql').insert().into(this._tableName)
```

Comment:
Yeah.  `ON DUPLICATE KEY UPDATE` is MySQL-specific and the `onDupUpdate` function isn't exposed on the `Squel` object without it.